sqlx-macros/src/derives/attributes.rs

```diff
@@ -91,7 +94,9 @@ pub fn parse_container_attributes(input: &[Attribute]) -> syn::Result<SqlxContai
                                 path,
                                 lit: Lit::Str(val),
                                 ..
-                            }) if path.is_ident("rename") => try_set!(rename, val.value(), value),
+                            }) if path.is_ident("type_name") => {
```

Comment:
For at least the next release, we should probably still match on `rename` but issue an error that it's been changed to `type_name`.

Reply:
If we had a decent, stable way to emit deprecation warnings I'd say we should accept both for the time being but alas the proc macro APIs haven't had any TLC in a long time.

Reply:
What about having the generated code contain a call to

```rust
#[doc(hidden)]
#[inline(always)]
#[deprecated = "sqlx(rename) is now called sqlx(type_name)"]
pub fn _rename() {}
```

Reply:
I think that's a good solution. It would contain a deprecated warning to the right place. 👍